python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -499,10 +523,14 @@ cdef class ParquetWriter:
                 table[name]._column, self.tbl_meta.get().column_metadata[i]
             )
 
-        pandas_metadata = generate_pandas_metadata(table, self.index)
+        index = (
+            False if isinstance(table._index, cudf.RangeIndex) else self.index
+        )
+        pandas_metadata = generate_pandas_metadata(table, index)
+        cdef map[string, string] tmp_user_data
+        tmp_user_data[str.encode("pandas")] = str.encode(pandas_metadata)
         cdef vector[map[string, string]] user_data
-        user_data.resize(1)
-        user_data.back()[str.encode("pandas")] = str.encode(pandas_metadata)
+        user_data = vector[map[string, string]](num_partitions, tmp_user_data)
```

Comment:
In the case of a partitioned write do we really want the pandas metadata on all partitions? Not sure I follow that.

Reply:
Each individual file needs to be a readable parquet file. The pandas metadata is in all the files generated by the `to_parquet(partition_cols)` which is modeled after the method of the same name from pandas. Same for the files generated by pandas's `df.to_parquet(partition_cols)`. So I suppose the chunked writer shouldn't be any different.